src/main/java/AaronBot.java

```diff
@@ -0,0 +1,275 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the main AaronBot class, executes the user-bot interaction task adding/deleting/marking
+ */
+public class AaronBot {
+    private static TaskListFile fileReader = new TaskListFile();
+    private static ArrayList<Task> taskList = new ArrayList<>();
+    public static void main(String[] args) {
+        try {
+            taskList = TaskListFile.loadFromFile("TaskList.txt");
+            if (taskList.size() > 0) {
+                System.out.println("Previous list found!");
+                for (int x = 0; x < taskList.size(); x++) {
+                    System.out.println((x + 1) + " " + taskList.get(x)); 
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println(e);
+        }
+        boolean isBye = true;
+        Scanner inputScanner = new Scanner(System.in);
+        System.out.println("Hello, I am AaronBot, please talk to me I love my students very much :)");
+        System.out.println("To add a task to the list, type 'add ' followed by your task.");
+        System.out.println("To add a Todo: add todo [task name]");
+        System.out.println("To add a Deadline: add deadline [task name] /by [deadline]");
+        System.out.println("To add an Event: add event [task name] /from [start] /to [end]");
+        System.out.println("To show the list, type 'show list'.");
+        System.out.println("To mark a task done: mark [index of task]");
+        System.out.println("To unmark a task done: unmark [index of task]");
+        while (isBye) {
+            try {
+                isBye = chat(inputScanner.nextLine());
+            } catch (NonsenseCommandException e) {
+                System.out.println("What are you saying dear student?\n" + e.toString());
+            }
+        }
+        inputScanner.close();
+    }
+
+    /**
+     * Main function that accepts an input from the user and executes a command based on it.
+     * @param userInput String input entered by user
+     * @return boolean value representing whether the chat should still be continued
+     * @throws NonsenseCommandException if command given by user is not recognized by Aaronbot
+     */
+    private static boolean chat(String userInput) throws NonsenseCommandException{
```

Comment:
consider using a plural noun instead of tokenizedUserInput as your string array contains multiple strings.